pkg/collector/container.go

```diff
@@ -98,3 +121,93 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 		LivenessProbe:   livenessProbe,
 	}
 }
+
+func getConfigContainerPorts(logger logr.Logger, cfg string) map[string]corev1.ContainerPort {
+	ports := map[string]corev1.ContainerPort{}
+	c, err := adapters.ConfigFromString(cfg)
+	if err != nil {
+		logger.Error(err, "couldn't extract the configuration")
+		return ports
+	}
+	ps, err := adapters.ConfigToReceiverPorts(logger, c)
+	if err != nil {
+		logger.Error(err, "couldn't build container ports from configuration")
+	} else {
+		for _, p := range ps {
+			truncName := naming.Truncate(p.Name, maxPortLen)
+			if p.Name != truncName {
+				logger.Info("truncating container port name",
+					"port.name.prev", p.Name, "port.name.new", truncName)
+			}
+			nameErrs := validation.IsValidPortName(truncName)
+			numErrs := validation.IsValidPortNum(int(p.Port))
+			if len(nameErrs) > 0 || len(numErrs) > 0 {
+				logger.Error(errInvalidPort, "dropping container port", "port.name", truncName, "port.num", p.Port,
+					"port.name.errs", nameErrs, "num.errs", numErrs)
+				continue
+			}
```

Comment:
I might be able to move this truncation+validation to inside the `adapters.ConfigToReceiverPorts()` function - what do you think, @pavolloffay?

Reply:
I tried to move it and it caused an import cycle. 😞 So I'm just going to leave the code as it is.